### PR TITLE
Internal: Floating IP cleanup

### PR DESCRIPTION
**Pull request by ozerovandrei**
Move Networking V2 networks and floating IPs helpers functions outside
of resource_openstack_networking_floatingip_v2.go.
Update debug and error messages.

For #456

### CODE REVIEW

**Comment by jtopjian** [comment]
@ozerovandrei With #570 merged, there's a lot of rebasing we're going to have to do :) 

And just a heads up, I have the following "cleanup" branches queued: identity, block storage snapshots, and firewall.

**Comment by ozerovandrei** [comment]
@jtopjian yeah, #570 was big.

Alright, I will grab networking-related stuff for now.

**Comment by jtopjian** [comment]
@ozerovandrei Looks like an easy way to handle the rebase is to do:

```
gofmt -s -w ./openstack/specific-file.go
```

Then commit and amend the changes to a previous commit. Then rebase with master.

You might still get merge conflicts, but they will be easier to handle.

File: openstack/resource_openstack_networking_floatingip_v2.go [hunk 1]
```diff
@@ -114,39 +121,40 @@ func resourceNetworkFloatingIPV2Create(d *schema.ResourceData, meta interface{})
...
+		return fmt.Errorf("Error creating openstack_networking_floatingip_v2: %s", err)
 	}
 
-	log.Printf("[DEBUG] Waiting for OpenStack Neutron Floating IP (%s) to become available.", floatingIP.ID)
+	log.Printf("[DEBUG] Waiting for openstack_networking_floatingip_v2 %s to become available.", fip.ID)
 
 	stateConf := &resource.StateChangeConf{
-		Target:     []string{"ACTIVE"},
-		Refresh:    waitForFloatingIPActive(networkingClient, floatingIP.ID),
+		Target:     []string{"ACTIVE", "DOWN"},
```
Comment (jtopjian) [inline openstack/resource_openstack_networking_floatingip_v2.go]: Let's keep this just `[]string{"ACTIVE"}`.

I consider this an "improvement" patch (and have a changelog entry). We can do a separate PR that modifies this and includes an acceptance test that tests for a state of `DOWN`.

File: openstack/resource_openstack_networking_floatingip_v2.go [hunk 2]
```diff
@@ -229,123 +239,24 @@ func resourceNetworkFloatingIPV2Delete(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("Error creating OpenStack network client: %s", err)
 	}
 
+	if err := floatingips.Delete(networkingClient, d.Id()).ExtractErr(); err != nil {
+		return CheckDeleted(d, err, "Error deleting openstack_networking_floatingip_v2")
+	}
+
 	stateConf := &resource.StateChangeConf{
-		Pending:    []string{"ACTIVE"},
+		Pending:    []string{"ACTIVE", "DOWN"},
```
Comment (jtopjian) [inline openstack/resource_openstack_networking_floatingip_v2.go]: Same here

File: openstack/resource_openstack_networking_floatingip_v2.go (hunk 1 above)
Comment (ozerovandrei) [reply openstack/resource_openstack_networking_floatingip_v2.go]: I changed it because `waitForFloatingIPActive()` returned "ACTIVE" when floating IP was in the "ACTIVE" or "DOWN" state.
So it's not an "improvement" patch, I just moved this logic outside of deleted `waitForFloatingIPActive()` function.

Comment (jtopjian) [reply openstack/resource_openstack_networking_floatingip_v2.go]: ah - yes, my mistake. Thank you :)

**Comment by ozerovandrei** [comment]
Rebased, I'll merge it after successful tests results.

**Comment by ozerovandrei** [comment]
recheck

**Comment by ozerovandrei** [comment]
@jtopjian I added some additional datasource-related changes.
